pkg/clientconn/useragent.go

```diff
@@ -0,0 +1,35 @@
+package clientconn
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strconv"
+
+	"github.com/stackrox/rox/pkg/version"
+)
+
+var userAgent string
+
+func init() {
+	SetUserAgent("stackrox")
+}
+
+// SetUserAgent formats and sets a value for the User-Agent for the process.
+func SetUserAgent(agent string) {
```

Comment:
Instead of relying on the processes to call `SetUserAgent`, we could do something similar to our singleton's for datastore's, meaning we use `once.Do()` within `GetUserAgent` to set the user agent once and afterwards just return it.

This way we don't have the dependency on the consumer on making sure to have called `SetUserAgent` beforehand. Otherwise, I would expect a comment for `GetUserAgent` stating explicitly that `SetUserAgent` has to be called beforehand, otherwise it will return an empty string.

Reply:
`GetUserAgent()` is called by packages which have no idea about the client binary, so the value has to be set on the process initialization. Default value is set to `stackrox`. I'll add a comment.